### PR TITLE
Dashboard & Project page

### PR DESCRIPTION
**Pull request by ranaprachi**
- Commit graph UI with multiple branches support
- Share modal UI Fix
- Clone Project UI
- Upload project UI
- List projects UI
- Delete project API integration
- Logout link and gravatar
- Implement of the tab functionality

### CODE REVIEW

**Comment by shresthagrawal** [comment]
Hello @ranaprachi, The amount of work you have done is such a small amount of time is remarkable! Awesome work! I am not a front-end dev, but visually your work looks quite solid. These are very minute changes that might even enhance the work:

* Closing the tabs should not change my current view
* Tab design looks different, in the original design there is a gray bar only in the first ink tab and then there are no more gray bars. 
* Layout becomes disproportional when the screen sized is changed. Though Adaptability was not there in the proposal. A little bit would be good to have. 
* In the original design the emails get separated by a box when you put a comma. This is very minute but would be good to have. 
* Sign, Push, Pull, Send Button don't have the same height as the original design
* The circles in the graph are not in perfect centre. (If this was intentionally please specify the reason)
* Arrows in the fork and merge of the graph are missing
* Branch background shade change and name is missing, also the background there has texture in the original design.
* "commit history" with the logo above the graph, looks odd and should be removed
* The screen proportion is not the same, left bar looks a little thinker
* The font in the project list page looks different. 

@falafeljan would be better in reviewing the actual code!

**Comment by ranaprachi** [comment]
> Hello @ranaprachi, The amount of work you have done is such a small amount of time is remarkable! Awesome work! I am not a front-end dev, but visually your work looks quite solid. These are very minute changes that might even enhance the work:
> 
> * Closing the tabs should not change my current view
- Can you explain this briefly what is the interaction you are looking for?
> * Tab design looks different, in the original design there is a grey bar only in the first ink tab and then there are no more grey bars.
- In the figma design only 2 tabs are shown but if we consider multiple tabs the differentiation is not very clear. Please refer the screenshot below. Let know if we should go ahead and make that change. 
Screenshot 2020-02-02 at 12 17 30 AM

> * Layout becomes disproportional when the screen sized is changed. Though Adaptability was not there in the proposal. A little bit would be good to have.
- can you further explain this over a call?
> * In the original design the emails get separated by a box when you put a comma. This is very minute but would be good to have.
- The figma design looks like a type-ahead suggestion list and in the sheet it was mentioned improving UI so we did not account for this change. In order to implement this we would need a email-name autocomplete list of users.
> * Sign, Push, Pull, Send Button don't have the same height as the original design
- Will fix this
> * The circles in the graph are not in perfect centre. (If this was intentionally please specify the reason)
- Will fix this
> * Arrows in the fork and merge of the graph are missing
- Will fix this
> * Branch background shade change and name is missing, also the background there has texture in the original design.
- Branch name details were not there in the mock data. Plus implementing the same would not be feasible right now in the current implementation as we had mentioned in the initial discussion.
> * "commit history" with the logo above the graph, looks odd and should be removed
- Will do this
> * The screen proportion is not the same, left bar looks a little thinker
- Will fix this
> * The font in the project list page looks different.
- I had used the same font throughout, will look into this. 
> @falafeljan would be better in reviewing the actual code!

**Comment by shresthagrawal** [comment]
> Can you explain this briefly what is the interaction you are looking for?
If I am in tab A and I close some tabs X. My tab should remain tab A.
If I am in Tab A and I close the Tab A itself. I should be in the Project list page.

**Comment by shresthagrawal** [comment]
> * In the figma design only 2 tabs are shown but if we consider multiple tabs the differentiation is not very clear. Please refer the screenshot below. Let know if we should go ahead and make that change.

The screenshot looks good for now

File: renderer/pages/home.js [hunk 1]
```diff
@@ -1,153 +1,81 @@
...
-    const { canceled, filePaths } = await remote.dialog.showOpenDialog({
-      properties: ['openDirectory'],
+  const [selectedTab, setSelectedTab] = React.useState(0);
+
+  const [tabs, setTabs] = React.useState([{
+    id: null,
+    head: () => (
+      <HeaderText color="#fff" size="16px" weight="900" family="Bowlby One">
+        INK
+      </HeaderText>
```
Comment (falafeljan) [inline renderer/pages/home.js]: i'd prefer to have this as an SVG, didn't we have one before?

File: renderer/components/Tabs.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                                            )
+                                    }
+                                    {index < tabs.length-1 ? <TabsHeadDivider/> : null}
+                                </TabHeaderContainer>
+                            </Space>
+                        ))}
+                    </FlexContainer>
+                </Size>
+                <TabBodyContainer>
+                    {tabs[selectedTab].content()}
```
Comment (falafeljan) [inline renderer/components/Tabs.js]: i'm receiving the following error after closing two tabs after each other:

grafik

**Comment by shresthagrawal** [comment]
LGTM! nice work @ranaprachi 
@falafeljan can you also quickly review and merge the PR.

**Comment by shresthagrawal** [comment]
@ranaprachi can you rebase the PR so that @falafeljan can merge it.

**Comment by falafeljan** [comment]
we'll continue to merge this work in another PR, #60! @ranaprachi, thank you for your fork.

**Comment by ranaprachi** [comment]
> @ranaprachi can you rebase the PR so that @falafeljan can merge it.

Sorry missed your comment.

**Pull request closed without merging**